packages/core/src/react-integration/hooks/useRetrieve.ts

```diff
@@ -9,8 +9,11 @@ import { DispatchContext } from '../context';
 export default function useRetrieve<Shape extends ReadShape<any, any>>(
   fetchShape: Shape,
   params: ParamsFromShape<Shape> | null,
+  trigger = false,
```

Comment:
again, trigger what?

Reply:
trigger teh retrieval, i.e., fetch. This part is kinda uncomfortable because it extends this hook interface when I'd prefer not to. Would love an alternative way - but essentially I need to make sure that when something is deleted its useResource() throws - which can only happen if useRetrieve() triggers a new fetch. maybe i should just layer another fetch if this one doesn't happen instead of conflating useretreive.

```typescript
maybePromise = useMemo(() => {
  // null params mean don't do anything
  if (!triggerFetch || !params | maybePromise) return;
  return fetch(params);
  // eslint-disable-next-line react-hooks/exhaustive-deps
}, [!!maybePromise, triggerFetch]);
```